components/ui/Inputs/PercentageInput.vue

```diff
@@ -1,10 +1,22 @@
 <template>
-  <t-input ref="inputRef" :value="formattedValue" />
+  <div>
+    <t-input
+      ref="inputRef"
+      type="number"
+      min="0"
+      max="100"
+      max-length="3"
+      :value="value"
+      @input="onInput"
+      @blur="onBlur"
+      @keypress.stop="
+        if (!/^[0-9.]$/i.test($event.key)) $event.preventDefault()
+      "
+    />
+  </div>
```

Comment:
without the extra <div> the linter moves the keypress validator onto the same line which errors in the compiler, i dont know how to prevent the linting